### ISSUE TITLE
$merge command still returns copy even if no modifications made

### ISSUE THREAD

**Issue by markwhitfeld** [kolodny/immutability-helper#35]
I think that the $merge command should be a bit more intelligent if the merges would not affect the state.
So if I merge an empty object, or an object with the same values it should not clone but rather return the original. For example the following results should return the original:

```javascript
let original = { a: { b: 'c' } };
let result1 = update(original, { a: { $merge: {} } } );
let result2 = update(original, { a: { $merge: { b: 'c' } } } );
```

I have published a running example in runkit:
https://runkit.com/markwhitfeld/immutability-helper-merge-example/1.0.0

In the second half you of this runkit sample you will see that I replace the $merge command with a new implementation that would give the desired behaviour by using the $set operator for each of the merged properties.

What do you think?

**Comment by kolodny** [comment]
Thanks for the feedback. I like the direction you're going with using `$set`. If you want to create a PR with tests I'll be happy to merge it in.

**Comment by kolodny** [comment]
This has been fixed on master via 9c0134d . Let me know if that solves the issue and I'll go ahead and publish it

**Comment by markwhitfeld** [comment]
I have just tested this and it passes all my tests. Please publish.
Thanks!
PS. It also sped up my test run fractionally over my $set property workaround.
